### ISSUE TITLE
`Exam mode`: Artemis may leak programming exercise titles to students

### ISSUE THREAD

**Issue by pogobanane** [ls1intum/Artemis#9121]
### Describe the bug

When changing an exercise description, students get notifications about the title of the exercise (not exercise group). This should not happen before an exam has started!

### To Reproduce

1. create student exams
1. We suspect that this happens **when the exam is already visible to students**, but has not started yet. 
2. Change the exercise description
3. Observe notifications as a student registered for the exam by navigating to the main exam in the exam tab and waiting a few seconds.

### Expected behavior

Exercise names should never be leaked to students before the exam started. 

### Screenshots

![grafik](https://github.com/user-attachments/assets/5aad3c88-b947-4302-abb0-c1a42b371f69)

### Which version of Artemis are you seeing the problem on?

7.4.4

### What browsers are you seeing the problem on?

Other (specify in "Additional context")

### Additional context

Briefly discussed with @az108

**Comment by az108** [comment]
A point to be added:

This only happens if the exams for students are already generated.

**Comment by Strohgelaender** [comment]
PR #9102 fixes this issue, notifications like this will only be shown after starting the exam.

**Comment by b-fein** [comment]
I think #9102 is only a partial fix that fixes the observable issue. The information leak still happens when looking at the network tab in the browser devtools. Ideally, the server would not send the information at all.

**Comment by Strohgelaender** [comment]
> Or does the live event button handle the requests internally and the request is not made in the first place if the component is no longer there?

I would assume so, but I have to double check. Good point, thank you!

**Comment by edkaya** [comment]
I still see the information in network tab even if the component is not present in the welcome page of the exam.

**Comment by edkaya** [comment]
I think it is quite important because the network tab also shows the problem statement itself even if the exam is not started and the instructor made some changes in problem statement. It would be nice to handle it on server side like not sending the response at all if the exam is not started.

**Comment by krusche** [comment]
I discussed this quickly internally today and assumed this information is passed to you @edkaya 

Here is my viewpoint:

1. Exams are not supposed to be visible multiple days before. Typically, they are visible a few hours before the exam starts when all information is correct. We should display a warning when instructors set the visible date too early. Let's say when there is more than 4h difference.

2. Students can click on start 5 min before the actual exam start, but the exam content is only shown to them when the actual exam starts.

3. Before this moment, there is no need to create exam live events for exercise updates at all!!! The server should simply do "nothing" here in terms of live events

4. In the short time window 5min before the exam (quite unlikely that the instructor changes something, but we should still cover it) until the exam officially starts, the update should be sent via websocket to the client and update the hidden information in the client, but should **not** be shown to the student, because he/she never looked at the problem statement, so there is no need to display something (which is annoying if it is not needed)

5. only after the exam started, the server should produce the actual exam live event and display it all students who are affected

**Comment by krusche** [comment]
@edkaya please implement it like that, potentially split into multiple PRs

The 5min time window extra handling would be low priority because it's very unlikely.

**Comment by edkaya** [comment]
Sure, I will have a look 👍
